### DISCUSSION TITLE
Convert meson dictionary or object from configuration_data() to json string

### DISCUSSION THREAD

**Discussion by aullidolunar** [mesonbuild/meson#13655] [Q&A] [answered]
This how I generate my dictionary:
```
cfg_data = configuration_data({
	'PROGRAM_NAME': PROGRAM_NAME,
	'PROGRAM_TITLE': PROGRAM_TITLE
})
```
I'm looking a way to pass the cfg_data as json string to my script:
```
run_target(
	'test-dict',
	command: [
		PYTHON_EXECUTABLE,
		SCRIPTS.get(6), # configure_file.py
		data_sources.get(0),
		'kakakaka.txt',
		cfg_data
	]
)
```
I try to use it in a run_target but:

> ERROR: run_target keyword argument 'command' was of type array[ConfigurationData | File | str] but should have been array[str | BuildTarget | CustomTarget | CustomTargetIndex | ExternalProgram | File] that cannot be empty

Any ideas?

**Comment by eli-schwartz** [comment, marked as answer]
I would try using https://mesonbuild.com/Reference-manual_functions.html#configure_file_output_format

Since meson 1.3.0, `configure_file(output: 'data.json', configuration: cfg_data, output_format: 'json')` can be used to configure a json file instead of a C header

**Comment by eli-schwartz** [comment]
Instead of json, if you only have two values you could pass them as values to the script, like:

```meson
	command: [
		PYTHON_EXECUTABLE,
		SCRIPTS.get(6), # configure_file.py
		data_sources.get(0),
		'kakakaka.txt',
		'--program-name', PROGRAM_NAME,
		'--program-title', PROGRAM_TITLE,
	]
```

**Reply by aullidolunar** [reply]
Thanks. In my real code will have more string to look up.

**Reply by eli-schwartz** [reply]
Yup, makes sense. I mentioned this possibility for completeness -- there's a reason I gave the other answer first. :)
